### PR TITLE
B t33543 export file

### PR DESCRIPTION
**Pull request by jalalifardemos**
**Ticket:** https://yaits.demos-deutschland.de/T33543

Description: fix the Error :   NameGenerator::$nameGenerator must not be accessed before initialization

### How to review/test
Test in Browser

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

**Review by mussbach — CHANGES_REQUESTED** [review]
I think you should rather inject the service in case it is missing

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/DocxResponseGenerator.php [hunk 1]
```diff
@@ -11,13 +11,15 @@
...
+use demosplan\DemosPlanCoreBundle\Logic\Procedure\NameGenerator;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpFoundation\StreamedResponse;
 
 class DocxResponseGenerator extends FileResponseGeneratorAbstract
 {
-    public function __construct(array $supportedTypes)
+    public function __construct(array $supportedTypes, NameGenerator $nameGenerator)
     {
+        parent::__construct($nameGenerator);
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/DocxResponseGenerator.php:22]: why do you need the parent call?

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/PdfResponseGenerator.php [hunk 2]
```diff
@@ -11,12 +11,14 @@
...
 use demosplan\DemosPlanCoreBundle\Exception\DemosException;
+use demosplan\DemosPlanCoreBundle\Logic\Procedure\NameGenerator;
 use Symfony\Component\HttpFoundation\Response;
 
 class PdfResponseGenerator extends FileResponseGeneratorAbstract
 {
-    public function __construct(array $supportedTypes)
+    public function __construct(array $supportedTypes, NameGenerator $nameGenerator)
     {
+        parent::__construct($nameGenerator);
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/PdfResponseGenerator.php:21]: why parent?

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/PdfResponseGenerator.php [hunk 3]
```diff
@@ -34,10 +36,12 @@ public function __invoke(array $file): Response
         $response = new Response($file['content'], 200);
         $response->headers->set('Pragma', 'public');
         $response->headers->set('Content-Type', 'application/pdf; charset=utf-8');
-        $response->headers->set(
-            'Content-Disposition',
-            $this->nameGenerator->generateDownloadFilename($file['filename'])
-        );
+        if ($this->nameGenerator) {
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/PdfResponseGenerator.php:39]: you can skip the if test when you inject the service

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/XlsResponseGenerator.php [hunk 4]
```diff
@@ -45,10 +47,12 @@ function () use ($writer) {
             'Content-Type',
             'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet; charset=utf-8'
         );
-        $response->headers->set(
-            'Content-Disposition',
-            $this->nameGenerator->generateDownloadFilename($file['filename'])
-        );
+        if ($this->nameGenerator) {
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/XlsResponseGenerator.php:50]: you can skip the if test when you inject the service

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/XlsResponseGenerator.php [hunk 5]
```diff
@@ -11,13 +11,15 @@
...
+use demosplan\DemosPlanCoreBundle\Logic\Procedure\NameGenerator;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpFoundation\StreamedResponse;
 
 class XlsResponseGenerator extends FileResponseGeneratorAbstract
 {
-    public function __construct(array $supportedTypes)
+    public function __construct(array $supportedTypes, NameGenerator $nameGenerator)
     {
+        parent::__construct($nameGenerator);
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/XlsResponseGenerator.php:22]: why?

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/DocxResponseGenerator.php (hunk 1 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/DocxResponseGenerator.php:22]: We have two __Construct and should call both. I think, we should either use parent or initialize parameter.

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/PdfResponseGenerator.php (hunk 3 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/PdfResponseGenerator.php:39]: Sorry, I forgot to remove that

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/DocxResponseGenerator.php (hunk 1 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/DocxResponseGenerator.php:22]: My reference was too old, it is correct here

**Review by mussbach — APPROVED** [review]
lgtm, thx. Please remember to squash the commits on merge

**Pull request merged**